packages/cli/yaml/validator/src/rules/valid-pagination/validateOffsetPagination.ts

```diff
@@ -151,23 +152,22 @@ function validateHasNextPageProperty({
     if (offsetPagination["has-next-page"] == null) {
         return [];
     }
-    return validateResponseProperty({
-        endpointId,
+    return validatePropertyInType({
         typeResolver,
         file,
-        responseProperty: offsetPagination["has-next-page"],
-        resolvedResponseType,
-        propertyValidator: {
-            propertyID: "has-next-page",
-            validate: isValidHasNextPageType
+        path: getPathFromSelector(offsetPagination["has-next-page"]),
```

Comment:
It looks like this would support any selector prefix - it doesn't actually validate that the user specified `$response`.